src/rust/src/wallet.rs

```diff
@@ -113,6 +114,9 @@ pub struct Wallet {
     /// The in-memory index from txid to notes from the associated transaction
     /// that have been decrypted with the IVKs known to this wallet.
     wallet_tx_notes: HashMap<TxId, TxNotes>,
+    /// The in-memory index from nullifier to the IDs for transactions
+    /// that attempt to spend the associated notes.
+    nullifiers: BTreeMap<OrderedNullifier, BTreeSet<TxId>>,
```

Comment:
I was surprised that this was a `BTreeSet` of `TxId`s rather than just a `TxId`. Why would we want more than one entry in this at any given time?

Reply:
The C++ wallet tracks conflicts between transactions. E.g. if the local wallet creates a transaction, adds it to the local mempool, and then a block arrives with a different transaction spending the same note. The wallet then has two transactions trying to spend the same note that conflict (only one can be in-chain at a time, and indeed in this example only one is). `zcash-cli gettransaction` has a `walletconflicts` field that lists the txids of wallet transactions that conflict with the one being queried, and the `BTreeSet<TxId>` here reflects that.

Reply:
If a transaction is unmined, then according to current zcashd wallet rules it can still be conflicted. So if you attempt to spend the same note in two different transactions from two separate wallets, they'll both be recorded here.